### PR TITLE
Add unhealthy host from 'lctl get_param health_check'

### PR DESCRIPTION
**Pull request by RDruon**
See [new snapshot](https://github.com/whamcloud/lustre-collector/blob/33870be05fb3aca647065648d737681ba41b9c85/src/snapshots/lustre_collector__parser__tests__node_output_unhealthy_host.snap#L38-L59)

### CODE REVIEW

File: src/top_level_parser.rs [hunk 1]
```diff
@@ -48,7 +96,11 @@ where
                 }
             }),
         ),
-        (param(HEALTH_CHECK), words().map(TopLevelStat::HealthCheck)),
+        // (param(HEALTH_CHECK), words().map(TopLevelStat::HealthCheck)),
```
Comment (jgrund) [inline src/top_level_parser.rs:99]: Cruft

**Review by jgrund — COMMENTED** [review]
Please review code before pushing to ensure all dead (commented) statements are removed.

File: src/top_level_parser.rs (hunk 1 above)
Comment (RDruon) [reply src/top_level_parser.rs:99]: Sorry about that, late night patches...

File: src/top_level_parser.rs [hunk 2]
```diff
@@ -25,7 +31,49 @@ enum TopLevelStat {
...
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        string("device"),
+        space(),
+        target(),
+        space(),
+        string("reported"),
+        space(),
+        words(),
```
Comment (jgrund) [inline src/top_level_parser.rs:49]: Use some `.skip(..)` here to reduce the size of the tuple needing destructuring.

Comment (jgrund) [inline src/top_level_parser.rs:49]: Use some `.skip(..)` here to reduce the size of the tuple needing destructuring.

Comment (jgrund) [inline src/top_level_parser.rs:49]: You are treating "health" as `words`. I'd feel more comfortable if you matched on the string explicitly.

You also should be sure that health only has that exact string as a posibility.

Comment (RDruon) [reply src/top_level_parser.rs:49]: Make sense, looks like it's always `unhealthy`
https://github.com/whamcloud/lustre/blob/a336d7c7c1cd62a5a5213835aa85b8eaa87b076a/lustre/obdclass/obd_sysfs.c#L524-L526

File: src/base_parsers.rs [hunk 3]
```diff
@@ -40,6 +40,7 @@ where
     many1(alpha_num().or(token('_')))
 }
 
+#[allow(dead_code)]
```
Comment (jgrund) [inline src/base_parsers.rs:43]: Is this not being used anywhere?

Comment (RDruon) [reply src/base_parsers.rs:43]: Not in the current code base:
```
warning: function `words` is never used
  --> src/base_parsers.rs:43:15
   |
43 | pub(crate) fn words<I>() -> impl Parser<I, Output = String>
   |               ^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: `lustre_collector` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 2.87s
```
But I haven't checked other PRs.

Comment (jgrund) [reply src/base_parsers.rs:43]: Let's remove it. I don't want to get into the habit of using linting rules to escape clippy.

**Review by jgrund — APPROVED** [review]

**Pull request merged**